packages/service-worker-mock/models/event-listeners.js

```diff
@@ -0,0 +1,53 @@
+// This is a derivate from https://github.com/GoogleChrome/workbox,
```

Comment:
Curious what the reasoning was for forking the listener code? Maybe @jeffposnick knows.

Reply:
I guess it's because workbox plugins can register event listeners and get handed down the `event` object and can individually call `event.waitUntil`, resulting in multiple promises to wait for, therefor the `_extendLifetimePromises` and `eventsDoneWaiting`.